vernac/declare.ml

```diff
@@ -1046,51 +1039,10 @@ let obligation_substitution expand prg =
   let ints = intset_to (pred (Array.length obls)) in
   obl_substitution expand obls ints
 
-let hide_obligation () =
-  Coqlib.check_required_library ["Coq"; "Program"; "Tactics"];
-  UnivGen.constr_of_monomorphic_global
-    (Coqlib.lib_ref "program.tactics.obligation")
```

Comment:
The gallina side https://github.com/coq/coq/blob/0a6444c522c18c634fe1030436ea82f326bada9a/theories/Program/Tactics.v#L323-L325 should also be removed.